bin/clinvar_jsons/traits_to_zooma_format.py

```diff
@@ -0,0 +1,218 @@
+import requests
+import sys
+from time import gmtime, strftime
+
+import progressbar
+
+from parse_trait_names import clinvar_jsons, get_traits_from_json
+from extract_pathogenic_and_likely_pathogenic_variants import has_allowed_clinical_significance
+
+
+DATE = strftime("%d/%m/%y %H:%M", gmtime())
+
+
+def main():
+    parser = ArgParser(sys.argv)
+
+    with open(parser.outfile_path, "wt") as outfile:
+        outfile.write("STUDY\tBIOENTITY\tPROPERTY_TYPE\tPROPERTY_VALUE\tSEMANTIC_TAG\tANNOTATOR\tANNOTATION_DATE\n")
+        line_count = file_len(parser.infile_path)
+        bar = progressbar.ProgressBar(max_value=line_count, widgets=[progressbar.AdaptiveETA(samples=1000)])
+        is_zooma_mapping_dict = {}
+        for clinvar_json in bar(clinvar_jsons(parser.infile_path)):
+            if not has_allowed_clinical_significance(clinvar_json):
+                continue
+            process_clinvar_json(clinvar_json, outfile, parser.zooma_host, parser.filters,
+                                 is_zooma_mapping_dict)
+
+
+def process_clinvar_json(clinvar_json, outfile, zooma_host, filters, is_zooma_mapping_dict):
+    clinvar_acc = get_clinvar_acc(clinvar_json)
+    variant_id_list = get_variant_ids(clinvar_json)
+
+    trait_dict = {}
+    trait_dict = get_traits_from_json(clinvar_json, trait_dict)
+
+    for variant_id, trait_dict_item in itertools.product(variant_id_list,
+                                                         trait_dict.items()):
+        trait_dict_key, trait_dict_value = trait_dict_item
+        trait_name = trait_dict_key
+        if trait_name == "not provided":
+            continue
+
+        if trait_name in is_zooma_mapping_dict:
+            is_zooma_mapping = is_zooma_mapping_dict[trait_name]
+        else:
+            zooma_uri_set = get_zooma_uris(trait_name, zooma_host, filters)
+            is_zooma_mapping = len(zooma_uri_set) > 0
+            is_zooma_mapping_dict[trait_name] = is_zooma_mapping
+
+        for xref in trait_dict_value.xref_set:
+            if xref.status.lower() != "current" or xref.db.lower() not in OntologyUri.db_to_uri_dict:
+                continue
+            ontology_uri = OntologyUri(xref.id_, xref.db)
+
+            if is_zooma_mapping:
+                continue
+
+            write_zooma_record(clinvar_acc, variant_id, trait_name, ontology_uri, DATE, outfile)
+
+
+def write_zooma_record(clinvar_acc, variant_id, trait_name, ontology_uri, date, outfile):
+    zooma_output_list = [clinvar_acc,
+                         variant_id,
+                         "disease",
+                         trait_name,
+                         str(ontology_uri),
+                         "clinvar-xrefs",
+                         date]
+    outfile.write("\t".join(zooma_output_list) + "\n")
+
+
+def get_variant_ids(clinvar_json):
+    rs_id_list = get_rs_ids(clinvar_json)
+    sv_id_list = get_sv_ids(clinvar_json)
+    return rs_id_list + sv_id_list
+
+
+def get_rs_ids(clinvar_json):
```

Comment:
I could combine this and the below function (get_sv_ids) to use just one loop and use an if-elif condition, i.e.:

```
def get_variant_ids(clinvar_json):
    variant_id_list = []
    for measure in clinvar_json["clinvarSet"]['referenceClinVarAssertion']["measureSet"]["measure"]:
        if "xref" in measure:
            for xref in measure["xref"]:
                if xref["db"].lower() == "dbsnp":
                    variant_id_list.append("rs{}".format(xref["id"]))
                elif xref["db"].lower() == "dbvar" and xref["id"].lower()[:3] in ("nsv", "esv"):
                    variant_id_list.append(xref["id"])
    return variant_id_list
```

which would save repeating code, considering it's not necessary to actually have separate functions for rs and sv ids. @cyenyxe , do you agree?